projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/facet/facet.component.ts

```diff
@@ -106,4 +114,63 @@ export class FacetComponent {
   getLinkParams(value: FacetValue) {
     return this.facetService.getLinkParams(value.query?.query?.value ?? '');
   }
+
+  onKeydown(event: KeyboardEvent): void {
+    // TODO: (CXSPA-6892) - Remove feature flag next major release.
+    if (!this.featureConfigService?.isEnabled('a11yFacetKeyboardNavigation')) {
+      return;
+    }
+    const targetIndex = this.values.toArray().findIndex((el) => {
+      return el.nativeElement === event.target;
+    });
+    switch (event.key) {
+      case 'ArrowLeft':
+        this.onArrowLeft(event);
+        break;
+      case 'ArrowRight':
+        this.onArrowRight(event);
+        break;
+      case 'ArrowDown':
+        this.onArrowDown(event, targetIndex);
+        break;
+      case 'ArrowUp':
+        this.onArrowUp(event, targetIndex);
+        break;
+    }
+  }
+
+  onArrowRight(event: Event): void {
+    if (this.isExpanded) {
```

Comment:
Not really anything wrong with this. Any argument for not putting the logic in the condition (inverted) rather than returning when the condition is met? Component logic is quite long so it could save some lines without being harder to read IMO.

Reply:
I've inverted the conditions. I've also added the feature flag to the template, restoring the default tab flow if the feature is off. I must have missed this earlier! ;)